sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/feed_range_composite_continuation_token.py

```diff
@@ -0,0 +1,173 @@
+                             f"[The {cls._continuation_property_name} must be non-empty array]")
+        continuation = [CompositeContinuationToken.from_json(child_range_continuation_token)
+                        for child_range_continuation_token in continuation_data]
+
+        # parsing feed range
+        feed_range: Optional[FeedRange] = None
+        if is_key_exists_and_not_none(data, FeedRangeEpk.type_property_name):
+            feed_range = FeedRangeEpk.from_json(data)
+        elif is_key_exists_and_not_none(data, FeedRangePartitionKey.type_property_name):
+            feed_range = FeedRangePartitionKey.from_json(data, continuation[0].feed_range)
+        else:
+            raise ValueError("Invalid feed range composite continuation token [Missing feed range scope]")
+
+        return cls(container_rid=container_rid, feed_range=feed_range, continuation=deque(continuation))
+
+    def handle_feed_range_gone(
+            self,
+            routing_provider: SmartRoutingMapProvider,
+            collection_link: str) -> None:
+        overlapping_ranges = routing_provider.get_overlapping_ranges(collection_link, [self._current_token.feed_range])
+
+        if len(overlapping_ranges) == 1:
+            # merge,reusing the existing the feedRange and continuationToken
+            pass
+        else:
+            # split, remove the parent range and then add new child ranges.
+            # For each new child range, using the continuation token from the parent
+            self._continuation.popleft()
+            for child_range in overlapping_ranges:
+                self._continuation.append(
+                    CompositeContinuationToken(
+                        Range.PartitionKeyRangeToRange(child_range),
+                        self._current_token.token))
+
+            self._current_token = self._continuation[0]
+
+    async def handle_feed_range_gone_async(
+            self,
+            routing_provider: AsyncSmartRoutingMapProvider,
+            collection_link: str) -> None:
+        overlapping_ranges = \
+            await routing_provider.get_overlapping_ranges(
+                collection_link,
+                [self._current_token.feed_range])
+
+        if len(overlapping_ranges) == 1:
+            # merge,reusing the existing the feedRange and continuationToken
+            pass
+        else:
+            # split, remove the parent range and then add new child ranges.
+            # For each new child range, using the continuation token from the parent
+            self._continuation.popleft()
+            for child_range in overlapping_ranges:
+                self._continuation.append(
+                    CompositeContinuationToken(
+                        Range.PartitionKeyRangeToRange(child_range),
+                        self._current_token.token))
+
+            self._current_token = self._continuation[0]
+
+    def should_retry_on_not_modified_response(self) -> bool:
```

Comment:
Is this isolated to full range or split scenarios? Is this code path possible when the customer wants to drain change feed for a partition key?

Reply:
yea it can happen for full range, split, or subset of ranges (like HPK)